Sources/MEWwalletTweetNacl/TweetNacl.swift

```diff
@@ -115,4 +124,27 @@ public class TweetNacl {
     
     return Data(m[Constants.SecretBox.zeroLength..<c.count])
   }
+    
+  public static func box(message: String, nonce: Data, theirPublicKey: Data, mySecretKey: Data) throws -> Data {
+    let k = try before(publicKey: theirPublicKey, secretKey: mySecretKey)
+    guard let data = message.data(using: .utf8) else { throw TweetNaclError.tweetNacl("Invalid message")}
+    return try secretbox(message: data, nonce: nonce, key: k)
+  }
+
+  private static func secretbox(message: Data, nonce: Data, key: Data) throws -> Data {
+    guard key.count == Constants.SecretBox.keyLength else { throw TweetNaclError.invalidKey }
+    guard nonce.count == Constants.SecretBox.nonceLength else { throw TweetNaclError.invalidNonce }
+      
+    var mData = Data(count: message.count + Constants.SecretBox.boxZeroLength)
+    mData.replaceSubrange(Constants.SecretBox.boxZeroLength..<mData.count, with: message)
```

Comment:
also `Constants.SecretBox.zeroLength`

Reply:
Which one does this refer to? Line 139? 
Should it be `mData.replaceSubrange(0..<Constants.SecretBox.boxZeroLength, with: message)`?

Reply:
Changed to `guard nonce.count == Constants.SecretBox.zeroLength...`

Reply:
@ronaldmannak 
> `mData.replaceSubrange(Constants.SecretBox.boxZeroLength..<mData.count, with: message)`

this part still incorrect, should be

`mData.replaceSubrange(Constants.SecretBox.boxZeroLength..<mData.count, with: message)`